components/support/sync-telemetry/src/main/java/mozilla/components/support/sync/telemetry/SyncTelemetry.kt

```diff
@@ -284,4 +287,42 @@ object SyncTelemetry {
         val message = reason.message ?: "Unexpected error: ${reason.code}"
         metric.set(message.take(MAX_FAILURE_REASON_LENGTH))
     }
+
+    fun processFxaTelemetry(jsonStr: String) {
+        val json = try {
+            JSONObject(jsonStr)
+        } catch (e: JSONException) {
+            logger.error("Invalid JSON in FxA telemetry", e)
+            return;
+        }
+        try {
+            val sent = json.getJSONArray("commands_sent")
+            for (i in 0..sent.length() - 1) {
+                val one = sent.getJSONObject(i)
+                val extras = mapOf(
+                    FxaTab.sentKeys.flowId to one.getString("flow_id"),
+                    FxaTab.sentKeys.streamId to one.getString("stream_id")
+                )
+                FxaTab.sent.record(extras)
+            }
+            logger.info("Reported telemetry for ${sent.length()} sent commands")
+        } catch (e: JSONException) {
+            logger.error("Failed to report sent commands", e)
+        }
+        try {
+            val recd = json.getJSONArray("commands_received")
+            for (i in 0..recd.length() - 1) {
+                val one = recd.getJSONObject(i)
+                val extras = mapOf(
+                    FxaTab.receivedKeys.flowId to one.getString("flow_id"),
+                    FxaTab.receivedKeys.streamId to one.getString("stream_id"),
+                    FxaTab.receivedKeys.reason to one.getString("reason")
+                )
+                FxaTab.received.record(extras)
+            }
+            logger.info("Reported telemetry for ${recd.length()} received commands")
+        } catch (e: JSONException) {
```

Comment:
maybe one big try/catch, with centralized error processing? i guess you don't get granular bits of telemetry in that case though..